tests/test_json_schema.py

```diff
@@ -5812,3 +5813,25 @@ class OuterModel(pydantic.BaseModel):
 
     with pytest.raises(ValidationError):
         OuterModel(x=2, y=-1, z=-1)
+
+
+def test_deprecated_fields():
```

Comment:
To address https://github.com/pydantic/pydantic/pull/8237/files#r1424563069, I would suggest:
```suggestion
@pytest.mark.skipif(
    pydantic.version.parse_mypy_version(importlib.metadata.version('typing_extensions')) < (4, 9),
    reason='deprecated type annotation requires typing_extensions>=4.9'
)
def test_deprecated_fields():
```
(And you could rename `parse_mypy_version` to `parse_package_version`, the function seems completely independent of package other than it requires the versioning scheme to be integers specified by `.`s (with an optional `+<anything>` on the end), which most packages do, including `mypy` and `typing_extensions` anyway.)